### PR TITLE
feat: make chain_id configurable

### PR DESCRIPTION
**Pull request by chirag-bgh**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #364 

## What is the new behavior?

-
-
-

### CODE REVIEW

File: crates/primitives/starknet/src/crypto/commitment/mod.rs [hunk 1]
```diff
@@ -202,8 +202,7 @@ where
     let nonce = FieldElement::from_bytes_be(&nonce.into()).unwrap();
     let version = FieldElement::from_byte_slice_be(&version.to_be_bytes()).unwrap();
     let tx_prefix = FieldElement::from_byte_slice_be(tx_prefix).unwrap();
-    // TODO: make it configurable
-    // FIXME: https://github.com/keep-starknet-strange/madara/issues/364
+
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/crypto/commitment/mod.rs:205]: take the chain id as argument so you don't have it hardcoded here

**Review by LucasLvy — CHANGES_REQUESTED** [review]
Small change otherwise looks good

File: crates/primitives/starknet/src/crypto/commitment/mod.rs (hunk 1 above)
Comment (chirag-bgh) [reply crates/primitives/starknet/src/crypto/commitment/mod.rs:205]: Then we might have to add `chain_id`  as input parameter here as well
https://github.com/keep-starknet-strange/madara/blob/151dffb9bb120e45163bf7ff3c5e366ce4b5f78a/crates/primitives/starknet/src/transaction/types.rs#L340

**Review by LucasLvy — CHANGES_REQUESTED** [review]
Change the chain id to be a`Felt252Wrapper` in the substrate storage. Create a getter 
like that (unsafe is fine here as it has to be an utf-8 string in the genesis config (if you think it's not then don't use the unsafe version):
```rust
    #[inline(always)]
    pub fn chain_id_str() -> String {
        unsafe { from_utf8_unchecked(&Self::chain_id().0.to_bytes_be()).to_string() }
    }
    ```
and the genesis config should take the chain id as a string `SN_GOERLI` for example and then convert it into a felt to save it in the storage. 
Let me know if that's unclear

**Review by LucasLvy — APPROVED** [review]
lgtm

**Pull request merged**